tests/integration/golang/helpers/s3.go

```diff
@@ -31,3 +35,51 @@ func NewS3Client(endpoint string) (*s3.Client, error) {
 		o.UsePathStyle = true
 	}), nil
 }
+
+func CreateBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		_, err := s3Client.CreateBucket(context.Background(), &s3.CreateBucketInput{
+			Bucket:                    aws.String(bucket),
+			CreateBucketConfiguration: &types.CreateBucketConfiguration{},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to create bucket '%s': %v", bucket, err)
+		}
+	}
+	return nil
+}
+
+func RemoveBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		// Delete all objects in the bucket
+		listObjectsOutput, err := s3Client.ListObjectsV2(context.Background(), &s3.ListObjectsV2Input{
```

Comment:
please use pagination

Reply:
for our integration test with 2 objects in each bucket?

Reply:
Arguably our integration tests should have an additional test case to validate listing large amounts of artifacts that don't fit in a single page, so yes

Reply:
that seems like a premature optimization for bridge we have not crossed, but I will make the change

Reply:
done